utils/get_supported_scylla_base_versions.py

```diff
@@ -0,0 +1,131 @@
+                if idx == 0:
+                    # The dest version is a minor release of latest enterprise release
+                    ent_base_version.append(version)
+                else:
+                    # Choose the last two releases as upgrade base
+                    ent_base_version += ent_release_list[idx-1:][:2]
+            elif version == 'enterprise' or parse_version(version) > parse_version(ent_release_list[0]):
+                oos_base_version.append(ent_release_list[-1])
+            elif re.match(r'\d+.\d+', version) and parse_version(version) >= parse_version(ent_release_list[0]):
+                oos_base_version.append(oos_release_list[-1])
+                ent_base_version += ent_release_list[-2:]
+        elif product == 'scylla':
+            if version in supported_versions:
+                # The dest version is a released opensource version
+                idx = oos_release_list.index(version)
+                if idx == 0:
+                    # The dest version is a minor release of latest enterprise release
+                    oos_base_version.append(version)
+                else:
+                    # Choose the last two releases as upgrade base
+                    oos_base_version += oos_release_list[idx-1:][:2]
+            elif version == 'master' or parse_version(version) > parse_version(oos_release_list[0]):
+                oos_base_version.append(oos_release_list[-1])
+            elif re.match(r'\d+.\d+', version) and parse_version(version) < parse_version(oos_release_list[0]):
+                # If dest version is smaller than the first supported opensource release, it might be an invalid dest version
+                oos_base_version.append(oos_release_list[-1])
+        else:
+            raise ValueError("Unsupported product %s" % product)
+
+        # Filter out unsupported version
+        oos_base_version = [i for i in oos_base_version if i in supported_versions]
+        ent_base_version = [i for i in ent_base_version if i in supported_versions]
+
+        return oos_base_version + ent_base_version
+
+    def get_version_list(self):
+        supported_versions = []
+
+        self.repo_maps = get_s3_scylla_repos_mapping(self.dist_type, self.dist_version)
+        # Filter out the unsupported versions
+        for version_prefix, _ in self.repo_maps.items():
+            if not re.match(r'\d+.\d+', version_prefix):
+                continue
+            if self.oos_start_support_version and not is_enterprise(version_prefix) and \
+                    parse_version(version_prefix) < parse_version(self.oos_start_support_version):
+                continue
+            if self.ent_start_support_version and is_enterprise(version_prefix) and \
+                    parse_version(version_prefix) < parse_version(self.ent_start_support_version):
+                continue
+            supported_versions.append(version_prefix)
+        click.echo(f'Supported Versions: {supported_versions}')
```

Comment:
Please move out click.echo to the sct.py

Reply:
fixed.